hazelcast-jet-sql/src/main/java/com/hazelcast/jet/sql/impl/JetPlan.java

```diff
@@ -534,4 +547,94 @@ public SqlResult execute(QueryId queryId) {
             return planExecutor.execute(this, queryId);
         }
     }
+
+    static class DeletePlan extends JetPlan {
```

Comment:
I would try to avoid depending on `SqlNode`s in here. Majority of this logic should be performed during validation & conversion. I would try to do that in a generic way so it could be reused later on.